A71-xds-fallback.md

```diff
@@ -127,94 +52,160 @@ to [xDS Protocol Specification][resource-does-not-exist].
-file. 
+This may happen either when setting up a new channel or if the xDS control
+plane becomes unavailable after a resource change notification was received.
+
+XdsClients will need to be changed to support multiple ADS connections for each
+authority. Once the fallback process begins, impacted XdsClient will establish
+a connection to the next xDS control plane listed in the bootstrap JSON. Then
+then XdsClient will subscribed to all watched resources on that server and will
+update the cache based on the received responses.
+
+Connecting to the lower-priority servers does not close gRPC connections to the
+higher-priority servers. XdsClient will still wait for xDS resources on the ADS
+stream. Once such resource is received, XdsClient will close connections to the
+lower-priority server(s) and will use the resources from the higher priority
+servers.
+
+### Potential issues
+
+#### Config tears
+
+Config tears happen when the client winds up using some combination
+of resources from the primary and fallback servers at the same time, even
+though that combination of resources was never validated to work together. In
+theory, this can cause correctness issues where we might send traffic to
+the wrong location or the wrong way, or it can cause RPCs to fail. Note
+that this can happen only when the primary and fallback server use the same
+resource names.
+
+This may also happen in non-fallback cases with the configs that span several
+xDS resources. The main case where this is likely to come up in practice is
+HTTP filter configs split across LDS and RDS. Control planes may avoid that
+problem by inlining the RouteConfiguration into the LDS resource.
+
+gRPC team considered addressing this in the fallback case by making the source
+xDS server name to the XdsClient but ultimately decided against it as it would
+require a significant implementation work and would not receive sufficient
+testing and may reduce reliability instead of increasing it. It would still not
+help with config tears in non-fallback cases.
+
+#### Flapping
 
-## Rationale
+Flapping is when the client switches over to a different server for LDS and
+RDS but then the server dies before it can send CDS, so the client switches
+back to the previous server and then re-updates LDS and RDS. This would cause
+the client to fail RPCs for a short period of time. Note that this can happen
+only when the primary and fallback server use different resource names.
 
-Other approaches considered:
+This will be addressed by moving the CDS and EDS watchers to the xDS resolver.
 
-### Using a single XdsClient instance for all data plane targets
+## Other approaches considered:
 
-Using a singleton XdsClient instance would cause individual channels that
-already have all of their resources cached from the primary server to switch
-to fallback resources in the following cases:
+### Unconditionally perform fallback when xDS control plane becomes unreachable
```

Comment:
These first two items seem to be conflating two independent things, which we should probably split up more cleanly.  I suggest describing them as follows:

1. Perform fallback based only on xDS server reachability, not taking into account whether there are watched resources that are not cached.  This was rejected because it would cause us to fallback even when everything is cached, which is undesirable.
2. Stick with a global XdsClient.  This was rejected because it would mean that if any one channel had a problem that triggered fallback, it would trigger fallback for *all* channels, even those that already have all of their resources cached.